src/libs/Navigation/AppNavigator/createCustomStackNavigator/CustomRouter.ts

```diff
@@ -1,78 +1,125 @@
-        name: NAVIGATORS.CENTRAL_PANE_NAVIGATOR,
+    const settingsHomeRouteName = mapScreenNameToSettingsScreenName[settingsScreenName] || SCREENS.SETTINGS.PROFILE.ROOT;
+
+    const fullScreenRoute = {
+        name: NAVIGATORS.FULL_SCREEN_NAVIGATOR,
         state: {
             routes: [
                 {
-                    name: SCREENS.REPORT,
-                    params: {
-                        reportID,
+                    name: SCREENS.SETTINGS.ROOT,
+                },
+                {
+                    name: SCREENS.SETTINGS_CENTRAL_PANE,
+                    state: {
+                        routes: [
+                            {
+                                name: settingsHomeRouteName,
+                            },
+                        ],
                     },
                 },
             ],
         },
     };
-    state.routes.splice(1, 0, centralPaneNavigatorRoute);
+    state.routes.splice(2, 0, fullScreenRoute);
     // eslint-disable-next-line no-param-reassign, @typescript-eslint/non-nullable-type-assertion-style
     (state.index as number) = state.routes.length - 1;
+    // eslint-disable-next-line no-param-reassign, @typescript-eslint/non-nullable-type-assertion-style
+    (state.stale as boolean) = true;
 };
 
 function CustomRouter(options: ResponsiveStackNavigatorRouterOptions) {
     const stackRouter = StackRouter(options);
 
     return {
         ...stackRouter,
-        getRehydratedState(partialState: StackNavigationState<ParamListBase>, {routeNames, routeParamList, routeGetIdList}: RouterConfigOptions): StackNavigationState<ParamListBase> {
+        getRehydratedState(partialState: StackNavigationState<RootStackParamList>, {routeNames, routeParamList, routeGetIdList}: RouterConfigOptions): StackNavigationState<ParamListBase> {
             const isSmallScreenWidth = getIsSmallScreenWidth();
             // Make sure that there is at least one CentralPaneNavigator (ReportScreen by default) in the state if this is a wide layout
-            if (!isAtLeastOneCentralPaneNavigatorInState(partialState) && !isSmallScreenWidth) {
-                // If we added a route we need to make sure that the state.stale is true to generate new key for this route
+            const topmostCentralPaneRoute = getTopmostCentralPaneRoute(partialState);
+            const topmostBottomTabRoute = getTopmostBottomTabRoute(partialState);
 
-                // eslint-disable-next-line no-param-reassign
-                (partialState.stale as boolean) = true;
+            const isBottomTabMatchingCentralPane = topmostCentralPaneRoute && TAB_TO_CENTRAL_PANE_MAPPING[topmostBottomTabRoute.name].includes(topmostCentralPaneRoute.name);
+
+            if (!isSmallScreenWidth && !isBottomTabMatchingCentralPane) {
+                // If we added a route we need to make sure that the state.stale is true to generate new key for this route
+                // @ts-expect-error Updating read only property
```

Comment:
Most likely this ts-expect-error can be avoided 🤔 Maybe try doing it with assertion?